indico/modules/events/contributions/client/js/index.jsx

```diff
@@ -29,28 +29,29 @@ import PublicationSwitch from './PublicationSwitch';
     if (!editableSubmissionButton) {
       return;
     }
-    const {
-      eventId,
-      contributionId,
-      contributionCode,
-      apiEditableSubmitUrl,
-      type,
-      uploadUrl,
-    } = editableSubmissionButton.dataset;
-
-    let response;
+
+    const availableTypes = JSON.parse(editableSubmissionButton.dataset.availableTypes);
+    const {eventId, contributionId, contributionCode} = editableSubmissionButton.dataset;
+
+    let responses;
     try {
-      response = await indicoAxios.get(fileTypesURL({confId: eventId, type}));
+      responses = await Promise.all(
+        availableTypes.map(type => indicoAxios.get(fileTypesURL({confId: eventId, type})))
+      );
     } catch (e) {
       handleAxiosError(e);
     }
-    const fileTypes = camelizeKeys(response.data);
+
+    const fileTypes = Object.assign(
```

Comment:
It is really a personal preference I think. `Object.fromEntries`, `_.fromPairs` or `Object.assign`. I think all of them are equally good.

Reply:
I think `Object.fromEntries` is still missing in Edge, otherwise I'd have suggested that. ;)

IMHO using fromEntries/fromPairs is be superior to `Object.assign(...)` since the former does exactly what you are doing here: creating a dict from a list of tuples

In any case, I think the `{},` isn't needed, is it?

Reply:
Yeah, but you can still polyfill it if you want :P And yeah `{}` isn't needed. It will use the first object as the "base" for the returned object (it is just that I want to be explicit about it).